pkg/collector/corechecks/cluster/orchestrator/processors/processor.go

```diff
@@ -160,36 +162,37 @@ func (p *Processor) Process(ctx *ProcessorContext, list interface{}) (processRes
 		resourceMetadataModels = append(resourceMetadataModels, resourceMetadataModel)
 
 		// Add resource manifest
-		resourceManifestModels = append(resourceManifestModels, &model.Manifest{
+		manifest := &model.Manifest{
 			Type:            int32(ctx.NodeType),
 			Uid:             string(resourceUID),
 			ResourceVersion: resourceVersion,
 			Content:         yaml,
 			Version:         "v1",
 			ContentType:     "json",
-		})
+		}
+		if ctx.BufferedManifestEnabled {
+			ctx.ManifestChan <- manifest
+		}
+		resourceManifestModels = append(resourceManifestModels, manifest)
 	}
 
 	// Chunking resources based on the serialized size of their manifest and maximum messages number
```

Comment:
for the sake of readability/keeping things similarly I would refactor this to `ChunkMetadata` so it matches below signature